subprojects/language-groovy/src/main/java/org/gradle/api/internal/tasks/compile/ApiGroovyCompiler.java

```diff
@@ -249,7 +249,10 @@ public boolean apply(File file) {
                         }));
 
                         try {
-                            javaCompiler.execute(spec);
+                            WorkResult javaCompilerResult = javaCompiler.execute(spec);
```

Comment:
Forked compilation with a custom executable is incremental too. So there's no difference there, it's just inherently less safe because we can miss certain source/class relationships in that case. Which is why we don't recommend using a custom javac executable.